actionpack/lib/action_dispatch/http/content_security_policy.rb

```diff
@@ -9,34 +9,32 @@ class Middleware
       POLICY = "Content-Security-Policy"
       POLICY_REPORT_ONLY = "Content-Security-Policy-Report-Only"
 
-      def initialize(app)
+      def initialize(app, warn_on_no_content_security_policy = false)
         @app = app
+        @warn_on_no_content_security_policy = warn_on_no_content_security_policy
       end
 
       def call(env)
         request = ActionDispatch::Request.new env
         _, headers, _ = response = @app.call(env)
 
-        return response unless html_response?(headers)
         return response if policy_present?(headers)
 
         if policy = request.content_security_policy
           nonce = request.content_security_policy_nonce
           nonce_directives = request.content_security_policy_nonce_directives
           context = request.controller_instance || request
-          headers[header_name(request)] = policy.build(context, nonce, nonce_directives)
+          headers[header_name(request)] = policy.build(context, nonce, nonce_directives, headers[CONTENT_TYPE])
+        else
+          if (Rails.env.development? || Rails.env.test?) && logger(request) && @warn_on_no_content_security_policy
```

Comment:
Checks for environment can't enter the action pack code. If we want different behavior per environment this should be a config on this class.